### PR TITLE
Add BlockingFunctionsConfig, AuthorizedDomains and QuotaConfig fields to identityplatform config

### PR DESCRIPTION
**Pull request by mraouffouad**
Add BlockingFunctionsConfig, AuthorizedDomains and QuotaConfig fields to identityplatform config

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement

```

### CODE REVIEW

File: mmv1/products/identityplatform/ProjectDefaultConfig.yaml
Comment (rainshen49) [inline mmv1/products/identityplatform/ProjectDefaultConfig.yaml:143]: Should this be a Timestamp, as in !ruby/object:Api::Type::Time

File: mmv1/products/identityplatform/ProjectDefaultConfig.yaml
Comment (tylerg-dev) [inline mmv1/products/identityplatform/ProjectDefaultConfig.yaml:140]: would it make sense for this to be a resource reference to https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloudfunctions_function instead of a pure string?  

In theory we could accommodate someone that uses Terraform for Auth but not Functions, but do we expect that to be a common use-case?

File: mmv1/products/identityplatform/ProjectDefaultConfig.yaml
Comment (rainshen49) [inline mmv1/products/identityplatform/ProjectDefaultConfig.yaml:145]: Should this also be output only?

File: mmv1/products/identityplatform/ProjectDefaultConfig.yaml
Comment (rainshen49) [inline mmv1/products/identityplatform/ProjectDefaultConfig.yaml:194]: This is a string based on https://cloud.google.com/identity-platform/docs/reference/rest/v2/Config#TemporaryQuota

File: mmv1/products/identityplatform/ProjectDefaultConfig.yaml
Comment (tylerg-dev) [inline mmv1/products/identityplatform/ProjectDefaultConfig.yaml:243]: are these keys represented as a resource in another product?  They aren't ApiKeys I know, but if we can reference an object that is managed it will be less error prone.

File: mmv1/products/identityplatform/ProjectDefaultConfig.yaml
Comment (mraouffouad) [reply mmv1/products/identityplatform/ProjectDefaultConfig.yaml:237]: Based on the GCIP team request, this field is removed from the PR for now.

File: mmv1/products/identityplatform/ProjectDefaultConfig.yaml
Comment (mraouffouad) [reply mmv1/products/identityplatform/ProjectDefaultConfig.yaml:243]: Ditto -- this field is removed from the PR for now.

File: mmv1/products/identityplatform/ProjectDefaultConfig.yaml
Comment (mraouffouad) [reply mmv1/products/identityplatform/ProjectDefaultConfig.yaml:140]: Would defer this to the reviewer to weigh in.

**Comment by mraouffouad** [comment]
@roaks3 This PR is ready for review.

File: mmv1/templates/terraform/examples/identity_platform_project_default_config.tf.erb
Comment (roaks3) [inline mmv1/templates/terraform/examples/identity_platform_project_default_config.tf.erb:22]: Due to https://github.com/hashicorp/terraform-provider-google/issues/13327, tests are currently disabled on this resource. In short, the initial apply (and later the destroy) that is done during tests will clear out other settings on the project.

I think there are two main directions you can go from here:
1. Keep the changes where they are, but keep in mind that this bug is unresolved, so your users may experience difficulties. You can attempt to resolve the bug if you choose, which would help make these new features more usable. For testing, you will need to turn off `skip_test` locally, and run the test in your own environment to ensure things are working.
2. Move the changes to `identityplatform/Config.yaml`, which appears to use the same endpoint but without this bug. For testing, you could probably remove the `skip_vcr: true` so that they run in the PR, otherwise you will need to run the test in your own environment.

**Review by roaks3 — CHANGES_REQUESTED** [review]
Apologies for the delay on this! The changes overall look good, I had a few small comments but nothing major.

The tests are not running for this resource, so to approve we will need you to show evidence of running the test locally. Alternatively, I mentioned an option to switch these fields to another resource where testing is enabled.

File: mmv1/templates/terraform/examples/identity_platform_project_default_config.tf.erb
Comment (rainshen49) [inline mmv1/templates/terraform/examples/identity_platform_project_default_config.tf.erb:42]: Since this will go into the docs, how about let's put the default domains here? This way even if they didn't visit Firebase's docs, they still have some clue to follow.

`authorized_domains = ["localhost", "${projectId}.firebaseapp.com", "${projectId}.web.app"]`

File: mmv1/products/identityplatform/ProjectDefaultConfig.yaml
Comment (mraouffouad) [reply mmv1/products/identityplatform/ProjectDefaultConfig.yaml:141]: Yep. Done.

File: mmv1/products/identityplatform/ProjectDefaultConfig.yaml
Comment (mraouffouad) [reply mmv1/products/identityplatform/ProjectDefaultConfig.yaml:134]: Done. And yes, ```forward_inbound_credentials``` is optional.

File: mmv1/templates/terraform/examples/identity_platform_project_default_config.tf.erb
Comment (rainshen49) [inline mmv1/templates/terraform/examples/identity_platform_project_default_config.tf.erb:44]: I was thinking more of doing it properly, like `"<%= ctx[:vars]['project_id'] %>"`, or actually put in a [data source](https://registry.terraform.io/providers/hashicorp/google-beta/latest/docs/data-sources/project) and reference.

File: mmv1/products/identityplatform/ProjectDefaultConfig.yaml
Comment (roaks3) [reply mmv1/products/identityplatform/ProjectDefaultConfig.yaml:181]: Was this addressed?

File: mmv1/templates/terraform/examples/identity_platform_project_default_config.tf.erb
Comment (roaks3) [reply mmv1/templates/terraform/examples/identity_platform_project_default_config.tf.erb:22]: I'm not sure this was addressed either. It seems like you're choosing to keep the changes in this resource (cc @tylerg-dev who I see is a reviewer and also authored `identityplatform/Config.yaml`  https://github.com/GoogleCloudPlatform/magic-modules/pull/6587).

Either way, you will need to provide some sort of proof (ie. output) from running the acceptance tests.

**Review by roaks3 — CHANGES_REQUESTED** [review]
I think a few things were not addressed, and we will need to confirm tests pass before approval

File: mmv1/templates/terraform/examples/identity_platform_project_default_config.tf.erb
Comment (tylerg-dev) [reply mmv1/templates/terraform/examples/identity_platform_project_default_config.tf.erb:22]: I can't see why we'd have a duplicate at all?  this one has more fields, but those should have just been merged into the initially added resource.  
At this point do we have a way to combine them back together (and properly use field-masks to avoid the bug listed above)

Comment (roaks3) [reply mmv1/templates/terraform/examples/identity_platform_project_default_config.tf.erb:22]: Yea, the second resources was added with https://github.com/GoogleCloudPlatform/magic-modules/pull/6679, and I believe it was created due to a miscommunication combined with unfortunate timing. Since users could be using the newer resource, we cannot simply remove or merge it, but we could choose to start down the deprecation process and remove it in the next major release.

Ideally, I think the fields on this resource should be moved to yours, and this resource removed.

**Pull request closed without merging**
